### PR TITLE
Move Keychron Q0 and Q0 Plus data-driven configuration; `keychron` keymap `rules.mk` cleanup

### PR DESCRIPTION
**Pull request by adophoxia**
## Description

Move most of the build options and LED Index configurations into `info.json` for the Keychron Q0 and Q0 Plus.
Removed unneeded `ENCODER_MAP_ENABLE` flag from `rules.mk` for `keychron` keymap for Keychron Q0
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* N/A

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/keychron/q0/rev_0131/info.json [hunk 1]
```diff
@@ -23,6 +37,66 @@
...
+            "pixel_rain": true,
+            "typing_heatmap": true,
+            "digital_rain": true,
+            "solid_reactive_simple": true,
+            "solid_reactive_multiwide": true,
+            "solid_reactive_multinexus": true,
+            "splash": true,
+            "solid_splash": true,
+            "keypresses": true,
+            "framebuffer_effects": true
```
Comment (drashna) [inline keyboards/keychron/q0/rev_0131/info.json:64]: pretty sure these aren't supported here.

Comment (adophoxia) [reply keyboards/keychron/q0/rev_0131/info.json]: Moved back into `config.h`

**Review by drashna — APPROVED** [review]

File: keyboards/keychron/q0/rev_0130/info.json [hunk 2]
```diff
@@ -6,15 +6,85 @@
...
     "diode_direction": "ROW2COL",
+    "bootmagic": {
+        "matrix": [0, 1]
+    },
     "processor": "STM32L432",
     "bootloader": "stm32-dfu",
+    "rgb_matrix": {
+        "driver": "CKLED2001",
+        "animations": {
+            "band_spiral_val": true,
```
Comment (tzarc) [inline keyboards/keychron/q0/rev_0130/info.json:37]:
```suggestion
            "band_spiral_val": true,
            "breathing": true,
```

File: keyboards/keychron/q0/rev_0131/info.json [hunk 3]
```diff
@@ -23,6 +37,64 @@
     },
     "processor": "STM32L432",
     "bootloader": "stm32-dfu",
+    "rgb_matrix": {
+        "driver": "CKLED2001",
+        "animations": {
+            "band_spiral_val": true,
```
Comment (tzarc) [inline keyboards/keychron/q0/rev_0131/info.json:43]:
```suggestion
            "band_spiral_val": true,
            "breathing": true,
```

**Review by tzarc — CHANGES_REQUESTED** [review]

File: keyboards/keychron/q0/rev_0130/info.json (hunk 2 above)
Comment (tzarc) [reply keyboards/keychron/q0/rev_0130/info.json:37]: Seemed to miss this.

File: keyboards/keychron/q0/rev_0131/info.json (hunk 3 above)
Comment (tzarc) [reply keyboards/keychron/q0/rev_0131/info.json:43]: Seemed to miss this.

File: keyboards/keychron/q0/rev_0130/info.json (hunk 2 above)
Comment (adophoxia) [reply keyboards/keychron/q0/rev_0130/info.json:37]: Fixed in recent commits.

File: keyboards/keychron/q0/rev_0131/info.json (hunk 3 above)
Comment (adophoxia) [reply keyboards/keychron/q0/rev_0131/info.json:43]: As above.

**Comment by tzarc** [comment]
Seems the LED positions are still unmatched
image

File: keyboards/keychron/q0/rev_0130/info.json [hunk 4]
```diff
@@ -6,15 +6,86 @@
...
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true,
+        "rgb_matrix": true,
+        "rgblight": false
+    },
     "diode_direction": "ROW2COL",
+    "bootmagic": {
+        "matrix": [0, 1]
+    },
```
Comment (tzarc) [inline keyboards/keychron/q0/rev_0130/info.json:31]:
```suggestion
```

Comment (adophoxia) [reply keyboards/keychron/q0/rev_0130/info.json:31]: Removed `"bootmagic":` from `info.json`

**Review by tzarc — APPROVED** [review]

**Pull request merged**